packages/gatsby/src/query/query-compiler.js

```diff
@@ -125,6 +125,7 @@ export const parseQueries = async ({
   let files = [
     path.join(base, `src`),
     path.join(base, `.cache`, `fragments`),
+    path.join(base, `.cache`, `head`),
```

Comment:
This makes Gatsby handle the `useStaticQuery` call in `cache-dir/head/index.js` (which is copied to `.cache/head/index.js` during all Gatsby commands). Without this, it doesn't work.